src/main/java/io/supertokens/webserver/InputParser.java

```diff
@@ -137,7 +137,12 @@ public static String parseStringOrThrowError(JsonObject element, String fieldNam
             if (!stringified.contains("\"")) {
                 throw new Exception();
             }
-            return ((JsonObject) element).get(fieldName).getAsString();
+            String s = element.get(fieldName).getAsString().trim();
+
+            if (s.matches("^(?=.{1,64}@)[A-Za-z0-9_-]+(\\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z]{2,})$")) {
```

Comment:
We use the following regex in our backend SDK: https://github.com/supertokens/supertokens-node/blob/master/lib/ts/recipe/emailpassword/utils.ts#L242. 

Might want to make this consistent with that.

Reply:
Done